app/config.py

```diff
@@ -1,23 +1,89 @@
-import os
-import pathlib
-
-from dotenv import load_dotenv
-
-load_dotenv()
-
-# Environment variables
-ELASTIC_PASSWORD = os.getenv("ELASTIC_PASSWORD")
-ELASTIC_USER = os.getenv("ELASTIC_USER")
-ELASTIC_URL = os.getenv("ELASTIC_URL")
-DSN_SENTRY = os.getenv("DSN_SENTRY")
-CURRENT_ENV = os.getenv("ENV", "dev")
-APM_URL = os.getenv("APM_URL")
-BASE_DIR = pathlib.Path(__file__).parent
-OPEN_API_PATH = BASE_DIR / "doc" / "open-api.yml"
-
-# Elastic APM configuration
-APM_CONFIG = {
-    "SERVICE_NAME": "SEARCH APM",
-    "SERVER_URL": APM_URL,
-    "ENVIRONMENT": CURRENT_ENV,
-}
+from functools import lru_cache
+from pathlib import Path
+from typing import Any
+
+from pydantic import AnyHttpUrl, Field, SecretStr, validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class APMConfig(BaseSettings):
+    url: AnyHttpUrl = Field(...)
+    service_name: str = Field("SEARCH APM")
+
+
+class DocsConfig(BaseSettings):
+    doc_path: Path = Field(
+        default_factory=lambda: Path(__file__).parent / "doc" / "open-api.yml"
+    )
+
+    @validator("doc_path")
+    def validate_path(cls, v):
+        if not v.exists():
+            raise ValueError(f"The specified path does not exist: {v}")
+        return v
+
+
+class ElasticConfig(BaseSettings):
+    password: SecretStr = Field(...)
+    user: str = Field(...)
+    url: AnyHttpUrl = Field(...)
+
+
+class RedisConfig(BaseSettings):
+    host: str = Field(default="redis")
+    port: str = Field(default="6379")
+    database: str = Field(default="0")
+
+
+class MatomoConfig(BaseSettings):
+    id_site: str = Field(...)
+    tracking_url: AnyHttpUrl = Field(...)
+
+
+class MetadataConfig(BaseSettings):
+    url_cc_json: AnyHttpUrl = Field(...)
+
+
+class SentryConfig(BaseSettings):
+    dsn: AnyHttpUrl = Field(...)
+
+    @validator("dsn")
+    def enforce_https(cls, v):
+        if v.scheme != "https":
+            raise ValueError("Sentry DSN must use HTTPS")
+        return v
+
+
+class Settings(BaseSettings):
+    model_config = SettingsConfigDict(
+        env_file=".env",
+        env_file_encoding="utf-8",
+        case_sensitive=False,
+        extra="ignore",
+        env_nested_delimiter="__",
+    )
+
+    apm: APMConfig = Field(...)
```

Comment:
In Pydantic, `Field(...)` is used to specify additional metadata or validation rules for a model field. The ellipsis (...) inside the Field() function represents that the field is required and does not have a default value. This means that when creating an instance of a Pydantic model, the field must be provided, otherwise, a validation error will be raised.